src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Helpers.cs

```diff
@@ -128,7 +128,7 @@ public static X509Certificate2 TryFindMatchingCertificate(this X509Certificate2C
 
         private static bool AreByteArraysEqual(byte[] ba1, byte[] ba2)
         {
-            if (ba1.Length != ba2.Length)
+            if (ba1 == null || ba2 == null || ba1.Length != ba2.Length)
```

Comment:
I made the method that returns the skid return null on the case of the implicit skid. Maybe returning Array.Empty<byte>() is better, or make EnvelopedCms check if the skid is null before doing comparison.